controllers/antreainstall_controller.go

```diff
@@ -39,133 +40,58 @@ import (
-		// Error reading the object - requeue the request.
-		log.Error(err, "Unable to retrieve Network.operator.openshift.io object")
-		return reconcile.Result{Requeue: true}, err
-	}
-
-	// Fetch antrea-install CR.
-	operConfig := &operatorv1.AntreaInstall{}
-	err = r.Client.Get(context.TODO(), types.NamespacedName{Namespace: operatortypes.OperatorNameSpace, Name: operatortypes.OperatorConfigName}, operConfig)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			msg := fmt.Sprintf("%s CR not found", operatortypes.OperatorConfigName)
-			log.Info(msg)
-			r.Status.SetDegraded(statusmanager.ClusterConfig, "NoAntreaInstallCR", msg)
-			return reconcile.Result{}, nil
-		}
-		log.Error(err, "failed to get antrea-install CR")
-		r.Status.SetDegraded(statusmanager.OperatorConfig, "InvalidAntreaInstallCR", fmt.Sprintf("Failed to get operator CR: %v", err))
-		return reconcile.Result{Requeue: true}, err
-	}
-	if request.Name == operConfig.Name && r.AppliedOperConfig != nil {
-		if reflect.DeepEqual(operConfig.Spec, r.AppliedOperConfig.Spec) {
-			log.Info("no configuration change")
-			return reconcile.Result{}, nil
-		}
-	}
+func (oc *AdaptorOc) SetupWithManager(r *AntreaInstallReconciler, mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&operatorv1.AntreaInstall{}).
+		Watches(&source.Kind{Type: &configv1.Network{}}, &handler.EnqueueRequestForObject{}).
+		Complete(r)
+}
 
+func applyConfig(r *AntreaInstallReconciler, config configutil.Config, clusterConfig *configv1.Network, operConfig *operatorv1.AntreaInstall, operatorNetwork *ocoperv1.Network) (reconcile.Result, error) {
 	// Fill default configurations.
-	if err = configutil.FillConfigs(clusterConfig, operConfig); err != nil {
+	if err := config.FillConfigs(clusterConfig, operConfig); err != nil {
 		log.Error(err, "failed to fill configurations")
 		r.Status.SetDegraded(statusmanager.OperatorConfig, "FillConfigurationsError", fmt.Sprintf("Failed to fill configurations: %v", err))
 		return reconcile.Result{Requeue: true}, err
 	}
 
 	// Validate configurations.
-	if err = configutil.ValidateConfig(clusterConfig, operConfig); err != nil {
+	if err := config.ValidateConfig(clusterConfig, operConfig); err != nil {
 		log.Error(err, "failed to validate configurations")
 		r.Status.SetDegraded(statusmanager.OperatorConfig, "InvalidOperatorConfig", fmt.Sprintf("The operator configuration is invalid: %v", err))
 		return reconcile.Result{Requeue: true}, err
 	}
 
 	// Generate render data.
-	renderData, err := configutil.GenerateRenderData(operatorNetwork, operConfig)
+	renderData, err := config.GenerateRenderData(operatorNetwork, operConfig)
 	if err != nil {
 		log.Error(err, "failed to generate render data")
 		r.Status.SetDegraded(statusmanager.OperatorConfig, "RenderConfigError", fmt.Sprintf("Failed to render operator configurations: %v", err))
 		return reconcile.Result{Requeue: true}, err
 	}
 
 	// Compare configurations change.
+	// TODO why not r.AppliedOperConfig?
```

Comment:
When operator restarts, r.AppliedOperConfig is empty, needs to get config from existing pods

Reply:
> When operator restarts, r.AppliedOperConfig is empty, needs to get config from existing pods

thanks danting, I'll remove the comment.